torch/_inductor/codegen/cpp.py

```diff
@@ -482,6 +482,17 @@ def merge_outer_fusion_kernels(
         self,
         cpp_kernel_proxy_list,
     ):
+        # Record the original parallel depth for each kernel
+        kernels_par_depth = [
+            cpp_kernel_proxy.decide_parallel_depth(
+                cpp_kernel_proxy.call_ranges[
+                    : cpp_kernel_proxy.loop_nest.max_parallel_depth()
+                ],
+                parallel_num_threads(),
+            )
+            for cpp_kernel_proxy in cpp_kernel_proxy_list
+        ]
```

Comment:
I feel it is simpler to just put this logic into `codegen_loops_impl` where the `par_depth` is computed for `OuterLoopFusedKernel`. Then, you don't have to add methods like `set_internal_kernels_par_depth` and `get_internal_kernels_par_depth`.